[Failing Test]: EnrichmentTest.test_enrichment_with_external_sqlserver flaky

### What happened?

https://github.com/apache/beam/actions/runs/20170681480/job/57905365353?pr=37092

```
    def test_enrichment_with_external_sqlserver(self, mock_stdout):
      try:
        db_adapter = DatabaseTypeAdapter.SQLSERVER
        with EnrichmentTestHelpers.sql_test_context(False, db_adapter):
          enrichment_with_external_sqlserver()
          output = mock_stdout.getvalue().splitlines()
          expected = validate_enrichment_with_external_sqlserver()
>         self.assertEqual(output, expected)
E         AssertionError: Lists differ: ["Row(product_id=3, name='C', quantity=10, region_id=4)", [106 chars]=1)"] != ["Row(product_id=1, name='A', quantity=2, region_id=3)", "[106 chars]=4)"]
E         
E         First differing element 0:
E         "Row(product_id=3, name='C', quantity=10, region_id=4)"
E         "Row(product_id=1, name='A', quantity=2, region_id=3)"
E         
E         - ["Row(product_id=3, name='C', quantity=10, region_id=4)",
E         -  "Row(product_id=1, name='A', quantity=2, region_id=3)",
E         ? ^
E         
E         + ["Row(product_id=1, name='A', quantity=2, region_id=3)",
E         ? ^
E         
E         -  "Row(product_id=2, name='B', quantity=3, region_id=1)"]
E         ?                                                        ^
E         
E         +  "Row(product_id=2, name='B', quantity=3, region_id=1)",
E         ?                                                        ^
E         
E         +  "Row(product_id=3, name='C', quantity=10, region_id=4)"]

apache_beam/examples/snippets/transforms/elementwise/enrichment_test.py:217: AssertionError
```

### Issue Failure

Failure: Test is flaky

### Issue Priority

Priority: 1 (unhealthy code / failing or flaky postcommit so we cannot be sure the product is healthy)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Infrastructure
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

--- a/sdks/python/apache_beam/examples/snippets/transforms/elementwise/enrichment_test.py
+++ b/sdks/python/apache_beam/examples/snippets/transforms/elementwise/enrichment_test.py
@@ -177,7 +177,7 @@ class EnrichmentTest(unittest.TestCase):
         enrichment_with_google_cloudsql_pg()
         output = mock_stdout.getvalue().splitlines()
         expected = validate_enrichment_with_google_cloudsql_pg()
-        self.assertEqual(output, expected)
+        self.assertEqual(sorted(output), sorted(expected))
     except Exception as e:
       self.fail(f"Test failed with unexpected error: {e}")
 
@@ -188,7 +188,7 @@ class EnrichmentTest(unittest.TestCase):
         enrichment_with_external_pg()
         output = mock_stdout.getvalue().splitlines()
         expected = validate_enrichment_with_external_pg()
-        self.assertEqual(output, expected)
+        self.assertEqual(sorted(output), sorted(expected))
     except (TestContainerStartupError, TestContainerTeardownError) as e:
       raise unittest.SkipTest(str(e))
     except Exception as e:
@@ -201,7 +201,7 @@ class EnrichmentTest(unittest.TestCase):
         enrichment_with_external_mysql()
         output = mock_stdout.getvalue().splitlines()
         expected = validate_enrichment_with_external_mysql()
-        self.assertEqual(output, expected)
+        self.assertEqual(sorted(output), sorted(expected))
     except (TestContainerStartupError, TestContainerTeardownError) as e:
       raise unittest.SkipTest(str(e))
     except Exception as e:
@@ -214,7 +214,7 @@ class EnrichmentTest(unittest.TestCase):
         enrichment_with_external_sqlserver()
         output = mock_stdout.getvalue().splitlines()
         expected = validate_enrichment_with_external_sqlserver()
-        self.assertEqual(output, expected)
+        self.assertEqual(sorted(output), sorted(expected))
     except (TestContainerStartupError, TestContainerTeardownError) as e:
       raise unittest.SkipTest(str(e))
     except Exception as e:

